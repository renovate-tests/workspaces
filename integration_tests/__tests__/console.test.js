/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 * @flow
 */

'use strict';

const skipOnWindows = require('../../scripts/skip_on_windows');
const {extractSummary} = require('../utils');
const runJest = require('../runJest');

skipOnWindows.suite();

test('console printing', () => {
  const {stderr, status} = runJest('console');
  const {summary, rest} = extractSummary(stderr);

  expect(status).toBe(0);
  expect(rest).toMatchSnapshot();
  expect(summary).toMatchSnapshot();
});

test('console printing with --verbose', () => {
  const {stderr, stdout, status} = runJest('console', [
    '--verbose',
    '--no-cache',
  ]);
  const {summary, rest} = extractSummary(stderr);

  expect(status).toBe(0);
  expect(stdout).toMatchSnapshot();
  expect(rest).toMatchSnapshot();
  expect(summary).toMatchSnapshot();
});

test('does not print to console with --silent', () => {
  const {stderr, stdout, status} = runJest('console', [
    // Need to pass --config because console test specifies `verbose: false`
    '--config=' +
      JSON.stringify({
        testEnvironment: 'node',
      }),
    '--silent',
    '--no-cache',
  ]);
  const {summary, rest} = extractSummary(stderr);

  expect(status).toBe(0);
  expect(stdout).toMatchSnapshot();
  expect(rest).toMatchSnapshot();
  expect(summary).toMatchSnapshot();
});
