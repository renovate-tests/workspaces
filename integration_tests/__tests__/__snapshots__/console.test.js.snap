// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`console printing 1`] = `
" PASS  __tests__/console.test.js
  ● Console

    console.log __tests__/console.test.js:11
      This is a log message.
    console.info __tests__/console.test.js:13
      This is an info message.
    console.warn __tests__/console.test.js:15
      This is a warning message.
    console.error __tests__/console.test.js:17
      This is an error message.


"
`;

exports[`console printing 2`] = `
"Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
"
`;

exports[`console printing with --verbose 1`] = `
"  console.log __tests__/console.test.js:11
    This is a log message.

  console.info __tests__/console.test.js:13
    This is an info message.

  console.warn __tests__/console.test.js:15
    This is a warning message.

  console.error __tests__/console.test.js:17
    This is an error message.

"
`;

exports[`console printing with --verbose 2`] = `
" PASS  __tests__/console.test.js
  ✓ works just fine

"
`;

exports[`console printing with --verbose 3`] = `
"Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
"
`;

exports[`does not print to console with --silent 1`] = `""`;

exports[`does not print to console with --silent 2`] = `
" PASS  __tests__/console.test.js

"
`;

exports[`does not print to console with --silent 3`] = `
"Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        <<REPLACED>>
"
`;
